sklearn/metrics/tests/test_pairwise.py

```diff
@@ -1549,3 +1550,14 @@ def test_numeric_pairwise_distances_datatypes(metric, global_dtype, y_is_x):
     dist = pairwise_distances(X, Y, metric=metric, **params)
 
     assert_allclose(dist, expected_dist)
+
+
+def test_sparse_manhattan_readonly_dataset():
+    # Non-regression test for: https://github.com/scikit-learn/scikit-learn/issues/7981
+    matrices1 = [csr_matrix(np.ones((5, 5)))]
+    matrices2 = [csr_matrix(np.ones((5, 5)))]
+    # Joblib memory maps datasets which makes them read-only.
+    # The following call was reporting as failing in #7981, but this must pass.
+    Parallel(n_jobs=-1, max_nbytes=0)(
```

Comment:
Let's not use `n_jobs=-1` in tests because running that test on a machine with 64 cores will start 64 Python interpreters that will import concurrently sklearn, numpy, scipy from disk which might slow down the machine for nothing (and make the test very slow to run).

```suggestion
    Parallel(n_jobs=2, max_nbytes=0)(
```